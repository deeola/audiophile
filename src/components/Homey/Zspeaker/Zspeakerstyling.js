import styled from "styled-components";
import bgpattern from "../../../assets/home/desktop/pattern-circles.svg";
import { Colors } from "../../../Themes";

//containers

export const ZspeakerConatiner = styled.div`
  background: ${Colors.orange};
  background-image: url(${bgpattern});
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  position: relative;
  margin: 40px auto;
  background-repeat: no-repeat;
  background-size: cover;
  overflow: hidden;
  background-size: 750px;

  @media (max-width: 802px) {
    background-size: center;
  }
`;

export const DescriptionImageWrapper = styled.div`
  width: 75%;
  margin: auto auto 0px auto;
  display: flex;
  justify-content: space-between;
  position: relative;
  padding-top: 120px;

  @media (max-width: 802px) {
    flex-direction: column;
    align-items: center;
    padding-top: 40px;
    width: 100%;
    justify-content: center;
  }
`;

export const Desc = styled.div`
  width: 40%;
  padding-bottom: 120px;

  @media (max-width: 802px) {
    width: 70%;
    padding-bottom: 50px;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction:column;
    margin-top: 50px;
  }

  @media (max-width: 490px) {
    width: 90%;
    margin-top: 40px;
  }
`;

export const Title = styled.h1`
  color: white;
  font-size: 56px;
  @media (max-width: 802px) {
    text-align: center;
    width: 50%;
  }

  @media (max-width: 490px) {
    width:70%;
  }
`;

export const Subtitle = styled.p`
  font-size: 15px;
  text-align: left;
  margin: 30px 0px;
  line-height: 25px;
  color: ${Colors.white};
  @media (max-width: 802px) {
    text-align: center;
    width: 70%;
  }

  @media (max-width: 490px) {
    width:80%;
    margin: 10px 0px;
  }
`;

//image

export const Zimage = styled.img`
  width: 41%;
  display: block;

  @media (max-width: 802px) {
    width: 30%;
  }
`;
