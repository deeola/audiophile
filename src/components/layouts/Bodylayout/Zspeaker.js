import React from "react";
import { Link } from "react-router-dom";
import { SecondaryButton } from "../../Buttons/Button";

function Zspeaker(props) {
  return (
    <div className="zspeakermain">
      <div className="zspeakerDiv">
        <p className="zspeakertitle">ZX7 SPEAKER</p>
        <SecondaryButton link={"/zx7"} label={'SEE PRODUCT'} />
      </div>
    </div>
  );
}

export default Zspeaker;


