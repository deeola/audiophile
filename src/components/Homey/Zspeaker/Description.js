import React from "react";
import { Desc, Subtitle, Title } from "./Zspeakerstyling";
import {BlackButton } from '../../Buttons/Button';

const Description = ({ title, subtitle }) => {
  return (
    <Desc>
      <Title data-testid='desctitle'>{title}</Title>
      <Subtitle data-testid='descsubtitle'>{subtitle}</Subtitle>
      <BlackButton  label={"SEE PRODUCT"} link='/zx9' />
    </Desc>
  );
};

export default Description;
