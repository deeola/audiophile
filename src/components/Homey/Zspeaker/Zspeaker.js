import React from "react";
import Description from "./Description";
import {
  Zimage,
  ZspeakerConatiner,
  DescriptionImageWrapper,
} from "./Zspeakerstyling";
import desktop from "../../../assets/home/desktop/speakerzx.png";
import tablet from "../../../assets/home/tablet/tabspeaker.png";
import mobile from "../../../assets/home/mobile/image-speaker-zx9.png";

const Zspeaker = () => {
  //use window width to deternine image source
  let src = desktop;

  if (window.innerWidth < 802) {
    src = tablet;
  } else if (window.innerWidth < 490) {
    src = mobile;
  }

  return (
    <ZspeakerConatiner>
      <DescriptionImageWrapper data-testid='descriptionimage'>
        <Zimage src={src} alt="photo" />
        <Description
          title={"ZX9 SPEAKER"}
          subtitle={
            "Upgrade to premium speakers that are phenomenally built to deliver truly remarkable sound."
          }
        />
      </DescriptionImageWrapper>
    </ZspeakerConatiner>
  );
};

export default Zspeaker;
