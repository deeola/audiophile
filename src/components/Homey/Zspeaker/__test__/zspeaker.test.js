import renderer from "react-test-renderer";
import { render } from "@testing-library/react";
import Zspeaker from "../Zspeaker";
import { BrowserRouter } from "react-router-dom";
import ReactDOM from "react-dom";

it("renders the div correctly", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <Zspeaker />
    </BrowserRouter>,
    div
  );
});

it("contains an image", () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <Zspeaker />
    </BrowserRouter>
  );

  expect(getByTestId("descriptionimage")).toContainHTML("img");
});

// create snapshot

it("matches snapshot", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Zspeaker />
      </BrowserRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
