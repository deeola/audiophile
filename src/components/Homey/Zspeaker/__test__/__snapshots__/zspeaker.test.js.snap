// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-bczRLJ uHPfx"
>
  <div
    className="sc-gsnTZi ivPMSB"
    data-testid="descriptionimage"
  >
    <img
      alt="photo"
      className="sc-jSMfEi iWRonm"
      src="speakerzx.png"
    />
    <div
      className="sc-dkzDqf hrzfoW"
    >
      <h1
        className="sc-hKMtZM ctHyQo"
        data-testid="desctitle"
      >
        ZX9 SPEAKER
      </h1>
      <p
        className="sc-eCYdqJ bQVPiA"
        data-testid="descsubtitle"
      >
        Upgrade to premium speakers that are phenomenally built to deliver truly remarkable sound.
      </p>
      <a
        className="sc-gKXOVf hXhkdj"
        href="/zx9"
        onClick={[Function]}
      >
        SEE PRODUCT
      </a>
    </div>
  </div>
</div>
`;
